ffi/capi/src/data_struct.rs

```diff
@@ -2,7 +2,8 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-#[cfg(feature = "icu_decimal")]
+#![cfg(feature = "icu_decimal")]
```

Comment:
I'll let @Manishearth approve this feature gate change

Reply:
without this there are dead code warnings in the harfbuzz tutorial

Reply:
If we want to feature-gate the whole module, it should probably be in lib.rs

If we want to feature-gate specific APIs, then feature-gate specific APIs

Reply:
It should not be on the module, because from the module's view, the `data_struct` module has nothing to do with decimal. All *current* methods on `ICU4XDataStruct` are behind that feature, so, I'd need to sprinkle the feature gate all over this file. Having it at the top is easier.

Reply:
Fine with this for now